dsa-23au/java-dsa/heaps-trees/src/main/BinarySearchTree.java

```diff
@@ -0,0 +1,19 @@
+import dev.codewithfriends;
+
+public class BinarySearchTree<T extends Comparable<? super T>> {
+    private static class BinaryNode<T> {}
+    private BinaryNode<T> root;
+    
+    public BinarySearchTree() {
+        root = null;
+    }
+
+    public void makeEmpty() {
```

Comment:
are these 2 methods just placeholders or do they have some other function?

Reply:
Those methods should be functional, they are right out of the Weiss book on pg. 114.
The first one is a constructor, and makeEmpty basically deletes the tree by making the root null.